### PR TITLE
feat(CDAP-19426): add parentversion in draft store

### PR DESCRIPTION
**Pull request by de-lan**
Add the column "parentversion" in draft store, the default value is null. made necessary changes to the draft apis.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/draft/Draft.java [hunk 1]
```diff
@@ -41,9 +41,9 @@ private Draft(ETLConfig config, String previousHash, String name, String descrip
 
   // This should be the default constructor until previousHash and revision are needed
   public Draft(ETLConfig config, String name, String description, ArtifactSummary artifact,
-               String id,
-               long createdTimeMillis, long updatedTimeMillis) {
-    this(config, "", name, description, 0, artifact, id, createdTimeMillis, updatedTimeMillis);
+               String id, long createdTimeMillis, long updatedTimeMillis, String parentVersion) {
+    this(config, "", name, description, 0, artifact,
```
Comment (sanjanasandeep) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/draft/Draft.java:45]: Should parentVersion be `@Nullable`?

File: cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DraftServiceTest.java [hunk 2]
```diff
@@ -365,12 +365,13 @@ private Draft createBatchPipelineDraft(DraftId draftId, String name, String desc
         .addConnection("src", "sink")
         .setEngine(Engine.SPARK)
         .build();
+    String parentVersion = "dummyParentVersion";
```
Comment (sanjanasandeep) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DraftServiceTest.java:369]: can we also do a test when parentVersion is null?

File: cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/draft/Draft.java (hunk 1 above)
Comment (de-lan) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/draft/Draft.java:45]: It is nullable, in the base Class `DraftStoreRequest`

File: cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DraftServiceTest.java (hunk 2 above)
Comment (de-lan) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DraftServiceTest.java:369]: updated a null test

**Review by sanjanasandeep — APPROVED** [review]
LGTM
